recipes-app/node-red-gpio/files/0005-Add-configurable-GPIO-mode-for-DIN.patch

```diff
@@ -0,0 +1,64 @@
+From c2ec2c88fc10f1d4f692ff7b655dc8a12758f09f Mon Sep 17 00:00:00 2001
+From: Ivan Mikhaylov <fr0st61te@gmail.com>
+Date: Thu, 28 Oct 2021 16:41:51 +0000
+Subject: [PATCH] Add configurable GPIO mode for DIN
+
+to close #135
+
+Signed-off-by: Ivan Mikhaylov <ivan.mikhaylov@siemens.com>
+---
+ hardware/intel/mraa-gpio-din.html | 13 ++++++++++++-
+ hardware/intel/mraa-gpio-din.js   |  3 ++-
+ 2 files changed, 14 insertions(+), 2 deletions(-)
+
+diff --git a/hardware/intel/mraa-gpio-din.html b/hardware/intel/mraa-gpio-din.html
+index d2ac00e..58a1363 100644
+--- a/hardware/intel/mraa-gpio-din.html
++++ b/hardware/intel/mraa-gpio-din.html
+@@ -7,7 +7,8 @@
+         defaults: {
+             name: {value:""},
+             pin:  {value:"", required: true},
+-            interrupt: {value:"", required: true}
++            interrupt: {value:"", required: true},
++            mode: {value:"", required: true}
+         },
+         inputs:0,
+         outputs:1,
+@@ -69,6 +70,16 @@
+             <option value="20">USER button</option>
+         </select>
+     </div>
++    <div class="form-row">
++        <label for="node-input-name"><i class="fa-level-up"></i> Mode</label>
++        <select type="text" id="node-input-mode" style="width: 250px;">
++            <option value='' disabled selected style='display:none;'>select mode</option>
++            <option value="0">Strong </option>
++            <option value="1">Pull-up </option>
++            <option value="2">Pull-down </option>
++            <option value="3">Hiz </option>
++        </select>
++    </div>
+     <div class="form-row">
+         <label for="node-input-name"><i class="fa-level-up"></i> Interrupt</label>
+         <select type="text" id="node-input-interrupt" style="width: 250px;">
+diff --git a/hardware/intel/mraa-gpio-din.js b/hardware/intel/mraa-gpio-din.js
+index 7eca35a..9fc3a12 100644
+--- a/hardware/intel/mraa-gpio-din.js
++++ b/hardware/intel/mraa-gpio-din.js
+@@ -7,10 +7,11 @@ module.exports = function(RED) {
+         RED.nodes.createNode(this,n);
+         this.pin = n.pin;
+         this.interrupt = n.interrupt;
++        this.mode = n.mode;
+         this.x = new m.Gpio(parseInt(this.pin));
+         this.board = m.getPlatformName();
+         var node = this;
+-        node.x.mode(m.PIN_GPIO);
++        node.x.mode(parseInt(this.mode));
```

Comment:
That was actually wrong before...

Reply:
Yes, it sets PIN_GPIO from pinmodes, not from GPIO modes.